src/gpio.rs

```diff
@@ -2,6 +2,12 @@
 
 use core::marker::PhantomData;
 
+#[cfg(feature = "std")]
+use std::boxed::Box;
```

Comment:
Just do the alloc thing, it also works with std enabled.

Reply:
Just did this, however it seems to require an explicit `extern crate alloc`.

Reply:
But that's not a problem, as long as the `extern crate alloc` declaration is also protected with `#[cfg(feature = "alloc")]`.